packages/react-composites/src/composites/MeetingComposite/adapter/AzureCommunicationMeetingAdapter.ts

```diff
@@ -14,12 +14,76 @@ import {
   IsScreenSharingOnChangedListener,
   DisplayNameChangedListener,
   IsSpeakingChangedListener,
-  CallAdapter
+  CallAdapter,
+  CallAdapterState
 } from '../../CallComposite';
-import { MessageReceivedListener, MessageReadListener, ChatAdapter } from '../../ChatComposite';
+import { MessageReceivedListener, MessageReadListener, ChatAdapter, ChatAdapterState } from '../../ChatComposite';
 import { MeetingAdapter, MeetingEvent } from './MeetingAdapter';
-import { MeetingAdapterState } from '../state/MeetingAdapterState';
-import { MeetingCompositePage } from '../state/MeetingCompositePage';
+import {
+  meetingAdapterStateFromBackingStates,
+  MeetingAdapterClientState,
+  MeetingAdapterState,
+  mergeCallAdapterStateIntoMeetingAdapterState,
+  mergeChatAdapterStateIntoMeetingAdapterState
+} from '../state/MeetingAdapterState';
+import { MeetingCompositePage, meetingPageToCallPage } from '../state/MeetingCompositePage';
+import { EventEmitter } from 'events';
+
+type MeetingAdapterStateChangedHandler = (newState: MeetingAdapterState) => void;
+
+/** Context of meeting, which is a centralized context for all state updates */
+class MeetingContext {
+  private emitter = new EventEmitter();
+  private state: MeetingAdapterState;
+
+  constructor(clientState: MeetingAdapterClientState) {
+    this.state = {
+      userId: clientState.userId,
+      displayName: clientState.displayName,
+      devices: clientState.devices,
+      meeting: undefined,
+      page: 'configuration'
+    };
+  }
+
+  public onStateChange(handler: MeetingAdapterStateChangedHandler): void {
+    this.emitter.on('stateChanged', handler);
+  }
+
+  public offStateChange(handler: MeetingAdapterStateChangedHandler): void {
+    this.emitter.off('stateChanged', handler);
+  }
+
+  public setState(state: MeetingAdapterState): void {
+    this.state = state;
+    this.emitter.emit('stateChanged', this.state);
+  }
+
+  public getState(): MeetingAdapterState {
+    return this.state;
+  }
+
+  public updateClientState(clientState: MeetingAdapterClientState): void {
+    this.setState({
+      userId: clientState.userId,
+      displayName: clientState.displayName,
+      meeting: clientState.meeting,
+      devices: clientState.devices,
+      page: this.state.page
+    });
+  }
+
+  public updateClientStateWithChatState(chatAdapterState: ChatAdapterState): void {
+    if (!this.state) {
+      console.warn('Cannot update chat state with meeting state - no meeting state exists');
+    }
+    this.updateClientState(mergeChatAdapterStateIntoMeetingAdapterState(this.state, chatAdapterState));
+  }
+
+  public updateClientStateWithCallState(callAdapterState: CallAdapterState): void {
+    this.updateClientState(mergeCallAdapterStateIntoMeetingAdapterState(this.state, callAdapterState));
```

Comment:
`if (!this.state)` check here not required?

Reply:
This is assigned in the constructor so always has a state, I'll remove the other `if(!this.state)` as it is not doing anything!